### ISSUE TITLE
fails

### ISSUE THREAD

**Issue by Artturin** [trofi/nix-olde#5]
```
$ cargo run -- --verbose
Fetching 'repology'
Fetching 'installed'
Fetching 'available'
Fetching from repology: ""
'available' failed, took 0.04 s.
'repology' failed, took 0.14 s.
'installed' done, took 17.21 s.

Error: MultipleErrors([CommandFailed { cmd: ["curl", "--compressed", "-s", "https://repology.org/api/v1/projects/?inrepo=nix_unstable&outdated=1"], output: Output { status: ExitStatus(unix_wait_status(15360)), stdout: "", stderr: "" } }, EmptyOutput("nix-env query")])
```
```
$ cargo run -- --nixpkgs $NIXGITS/my-nixpkgs --verbose
Fetching 'repology'
Fetching 'installed'
Fetching from repology: ""
Fetching 'available'
'repology' failed, took 0.12 s.
'available' done, took 14.85 s.
'installed' done, took 23.01 s.

Error: MultipleErrors([CommandFailed { cmd: ["curl", "--compressed", "-s", "https://repology.org/api/v1/projects/?inrepo=nix_unstable&outdated=1"], output: Output { status: ExitStatus(unix_wait_status(15360)), stdout: "", stderr: "" } }])
```

**Comment by Artturin** [comment]
not a issue with nix-env because
```diff
diff --git a/src/available.rs b/src/available.rs
index 4dca23c..ba6bd6d 100644
--- a/src/available.rs
+++ b/src/available.rs
@@ -115,6 +115,8 @@ pub(crate) fn get_packages(nixpkgs: &Option<String>) -> Result<BTreeSet<Package>
     let ps: BTreeMap<String, Available> =
         serde_json::from_slice(ps_u8.as_slice()).expect("valid json");
 
+    println!("{:?}", ps.keys());
+
     let r: BTreeSet<_> = ps
         .iter()
         .filter_map(|(attr, a)| {
```
prints attrs

**Comment by trofi** [comment]
Yeah, I think `cargo run` does some kind of sandbox or environment filtering to fail the HTTPS. A bit of debugging:

```diff
--- a/src/repology.rs
+++ b/src/repology.rs
@@ -47,7 +47,7 @@ pub(crate) fn get_packages(
         if verbose {
             eprintln!("Fetching from repology: {:?}", suffix);
         }
-        let contents_u8 = run_cmd(&["curl", "--compressed", "-s", &url])?;
+        let contents_u8 = run_cmd(&["curl", "--compressed", "-v", &url])?;
         // {
         //   "python:networkx": [
         //     {
```

```
cargo run -- --verbose
   Compiling nix-olde v0.1.0 (/home/slyfox/dev/nix-olde)
    Finished dev [unoptimized + debuginfo] target(s) in 0.47s
     Running `target/debug/nix-olde --verbose`
Fetching 'installed'
Fetching 'repology'
Fetching 'available'
Fetching from repology: ""
'repology' failed, took 0.09 s.
'installed' done, took 5.42 s.
'available' done, took 14.16 s.

Error: MultipleErrors([CommandFailed { cmd: ["curl", "--compressed", "-v", "https://repology.org/api/v1/projects/?inrepo=nix_unstable&outdated=1"], output: Output { status: ExitStatus(unix_wait_status(15360)), stdout: "", stderr: "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying [2a0a:2b41:94:cb5e::]:443...\n* Connected to repology.org (2a0a:2b41:94:cb5e::) port 443 (#0)\n* ALPN: offers h2,http/1.1\n} [5 bytes data]\n* TLSv1.3 (OUT), TLS handshake, Client hello (1):\n} [512 bytes data]\n*  CAfile: none\n*  CApath: /etc/pki/tls/certs\n{ [5 bytes data]\n* TLSv1.3 (IN), TLS handshake, Server hello (2):\n{ [122 bytes data]\n* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):\n{ [19 bytes data]\n* TLSv1.3 (IN), TLS handshake, Certificate (11):\n{ [3967 bytes data]\n* TLSv1.3 (OUT), TLS alert, unknown CA (560):\n} [2 bytes data]\n* SSL certificate problem: unable to get local issuer certificate\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n* Closing connection 0\ncurl: (60) SSL certificate problem: unable to get local issuer certificate\nMore details here: https://curl.se/docs/sslcerts.html\n\ncurl failed to verify the legitimacy of the server and therefore could not\nestablish a secure connection to it. To learn more about this situation and\nhow to fix it, please visit the web page mentioned above.\n" } }])
```

**Comment by trofi** [comment]
Disabled `-s` by default in https://github.com/trofi/nix-olde/commit/742b9dd9691781c2d5308d76e0e777ec73e97890

**Comment by Artturin** [comment]
```
./result/bin/nix-olde --nixpkgs $NIXGITS/my-nixpkgs
```
works but no --nixpkgs fails

```
$ ./result/bin/nix-olde
Fetching 'repology'
Fetching 'available'
Fetching 'installed'
'available' failed, took 0.04 s.
'repology' failed, took 2.19 s.
'installed' done, took 16.60 s.

Error: MultipleErrors([Canceled("Repology fetch"), EmptyOutput("nix-env query")])
```

found the issue, need ot resolve a another /etc/nixos, pr coming
